packages/luals-annotations/src/annotationBuilder.ts

```diff
@@ -0,0 +1,127 @@
+import luaLSType, { formatProperty } from "./luaLSType";
+import { Function, Property } from "@lest/docs";
+
+type FunctionRenderOptions = {
+	staticMethod?: boolean;
+	className?: string;
+};
+
+function renderParameterAnnotations(parameters: Property[]): string[] {
+	return parameters.map((param) => `---@param ${formatProperty(param, " ")}`);
+}
+
+function renderReturnAnnotations(returns: Property[]): string[] {
+	return returns.map((ret) => `---@return ${luaLSType(ret)} ${ret.name ?? ""}`);
+}
+
+function renderFunctionAliases(func: Function, staticMethod: boolean, className: string): string[] {
+	const { parameters = [] } = func;
+
+	const functionCharacter = staticMethod ? "." : ":";
+	const functionPrefix = className ? `${className}${functionCharacter}` : "";
+
+	const functionNames = [func.name, ...(func.aliases ?? [])];
+
+	return functionNames.map(
+		(name) => `function ${functionPrefix}${name}(${parameters.map((param) => param.name)}) end`
+	);
+}
+
+function renderDescription(description: string | string[]): string[] {
+	const descriptionLines = Array.isArray(description) ? description : [description];
+	return descriptionLines.map((line) => `--- ${line}`);
+}
+
+function renderFunctionDeclaration(
+	func: Function,
+	{ staticMethod = false, className = "" }: FunctionRenderOptions = {}
+): string {
+	const { parameters = [], returns = [], description = [] } = func;
+
+	// LuaLS has a very strict order about the function annotations, so this is why we have a specific order.
+	// 1. Description
+	// 2. Parameters
+	// 3. Returns
+	// 4. Function signature
+	const descriptionLines = renderDescription(description);
+	const paramList = renderParameterAnnotations(parameters);
+	const returnList = renderReturnAnnotations(returns);
+	const functionSignatures = renderFunctionAliases(func, staticMethod, className);
+
+	return functionSignatures
+		.map((signature) => [...descriptionLines, ...paramList, ...returnList, signature].join("\n"))
+		.join("\n");
+}
+
+export class DocumentBuilder {
+	lines: string[] = [];
+
+	add(line: string) {
+		this.lines.push(line);
+	}
+
+	build(): string {
+		return this.lines.join("\n");
+	}
+
+	buildToLines(): string[] {
+		return this.lines;
+	}
```

Comment:
What's the use of this function? Can you not just access `lines`?

Reply:
Change from a while ago, but the interface was already being depended upon so.. I kind of half-assed it.